### DISCUSSION TITLE
Filter a list

### DISCUSSION THREAD

**Discussion by robertjmccarthy** [ministryofjustice/moj-frontend#717] [RELEASED: Components, pages and patterns]
The [filter a list pattern component](https://design-patterns.service.justice.gov.uk/patterns/filter-a-list) is live in the MoJ Design System.

Use this discussion to suggest changes and improvements to this component.

**Comment by emma-nicol** [comment]
Has anyone used a date picker in a filter or know of a reason why we shouldn't use one?

**Comment by willemvangerwen-hippo** [comment]
Is there any rationale behind using links (<a>) for the column headings rather than <button>? From a semantics and accessibility point of view, button would seem like the better choice.

**Comment by chrispymm** [comment]
DfE recently shared some work they did around filtering:

- [Improving how candidates can filter and sort courses](https://becoming-a-teacher.design-history.education.gov.uk/find-teacher-training/improving-filtering-and-sorting/)
- [Researching filters across government](https://becoming-a-teacher.design-history.education.gov.uk/find-teacher-training/researching-filters-across-government/)
- [User research with candidates on filtering and sorting courses](https://becoming-a-teacher.design-history.education.gov.uk/find-teacher-training/filtering-and-sorting-ur/)

**Comment by manalishi79** [comment]
<img width="758" height="491" alt="Screenshot 2026-03-20 at 13 45 15" src="https://github.com/user-attachments/assets/9d6069db-d906-43fd-817d-761989ab455c" />
Is there a glitch in the MOJ DS website preview. 

https://design-patterns.service.justice.gov.uk/patterns/filter-a-list/

The show filter button is appearing on the right with a shadow running down the full length right side.

**Comment by ethan-mills-ukef** [comment]
We had a DAC report on a page that used this component with the following:

> The aria-haspopup attribute has been used on the filter button, which caused some 
confusion for our screen reader analyst. This is because this specific attribute tells the user 
that the button exposes a menu, however, it exposes a separate interactive layer not a 
menu.

> “I located a button on the page which read to me as ‘Show filters’ which once selected 
updated to read as ‘Hide filters’. The item announced to me as a menu and when selected 
made an audible noise as though a menu was expanding. When I attempted to close the 
menu, my focus initially became trapped on the button when I attempted to use the down 
cursor key meaning I needed to use additional keyboard commands to leave the button.
As the button behaves as an accordion, I would not usually expect to locate any additional 
menu presentation as this can affect screen reader users.
Removing the additional presentation within the code to present the buttons as a standard 
accordion will prevent any difficulty. This was consistent with JAWS and NVDA.”

> Solution:
You could either remove the aria-haspopup attribute completely or consider using it with 
the value of ‘dialog’ to inform screen reader users that a separate interactive layer will 
appear.
It is also best practice to use the aria-controls attribute alongside the aria-haspopup 
attribute as this creates a programmatic relationship between the button and the 
interactive layer the button controls

Does anyone have thoughts on this, or had similar? 
I don't believe we have any control over the "Show filter" button as a user of this component?
